lib/pubsub/subscription.js

```diff
@@ -296,32 +300,53 @@ Subscription.prototype.delete = function(callback) {
  *   // message.id = ID used to acknowledge its receival.
  *   // message.data = Contents of the message.
  * });
+ *
+ * //-
+ * // Limit the results.
+ * //-
+ * subscription.pull({ maxCount: 3 }, function(err, messages) {});
  */
 Subscription.prototype.pull = function(options, callback) {
   var that = this;
-  // TODO(jbd): Should not be racing with other pull.
+
   if (!callback) {
     callback = options;
     options = {};
   }
+
+  var batch = util.is(options.maxCount, 'number');
```

Comment:
Yes, this is correct. We can document this as a breaking change and help users through the migration if they require. My thought is most people will very welcome the change and like that we are efficiently returning all their messages at the same time.